libraries/botbuilder-ai/src/qnaMakerDialog.ts

```diff
@@ -199,14 +266,13 @@ export class QnAMakerDialog extends WaterfallDialog implements QnAMakerDialogCon
         endpointKey?: string,
         hostname?: string,
         noAnswer?: Activity,
-        threshold = 0.3,
-        activeLearningCardTitle = 'Did you mean:',
-        cardNoMatchText = 'None of the above.',
-        top = 3,
+        threshold?: number,
+        activeLearningCardTitle?: string,
+        cardNoMatchText?: string,
+        top?: number,
         cardNoMatchResponse?: Activity,
         strictFilters?: QnAMakerMetadata[],
         dialogId = 'QnAMakerDialog',
-        strictFiltersJoinOperator = JoinOperator.AND
```

Comment:
I think this could break existing code. If someone constructs an instance that uses every parameter, removing a parameter means existing code will fail to compile. Perhaps add `...args: unknown[]` to the constructor signature and then `if (args.length) console.warn('Deprecated parameter warning message...')`

Reply:
I would keep this parameter without default value since we don't have this parameter in C#.